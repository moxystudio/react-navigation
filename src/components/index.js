export { default as Navbar } from './navbar';
export { default as Drawer } from './drawer';
