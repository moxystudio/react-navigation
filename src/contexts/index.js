export { default as NavigationContext } from './NavigationContext';
