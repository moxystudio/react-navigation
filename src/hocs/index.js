export { default as withNavigation } from './withNavigation';
