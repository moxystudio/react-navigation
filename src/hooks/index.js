export { default as useDocumentScrollThrottled } from './useDocumentScrollThrottled';
