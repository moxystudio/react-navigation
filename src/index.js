// Components
export { Backdrop, Drawer, Navbar } from './components';

// Providers
export { NavigationProvider } from './providers';

// Hooks
export { useNavigation } from './hooks';

// HOCs
export { withNavigation } from './hocs';
