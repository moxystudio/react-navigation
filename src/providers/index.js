export { default as NavigationProvider } from './NavigationProvider';
