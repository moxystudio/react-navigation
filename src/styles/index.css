@import "./navbar.css";
@import "./drawer.css";
