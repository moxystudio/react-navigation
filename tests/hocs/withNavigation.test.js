import React from 'react';
import { render } from '@testing-library/react';
import stringifyObject from '../utils/stringify-object';
import { NavigationProvider, withNavigation } from '../../src';

const MyTestComponent = withNavigation()(({ navigation }) => (
    JSON.stringify(stringifyObject(navigation))
));

const renderWithProps = (props = {}) => render(
    <NavigationProvider>
        <MyTestComponent { ...props } />
    </NavigationProvider>,
);

describe('withNavigation HOC', () => {
    it('should provide navigation', () => {
        const { asFragment } = renderWithProps();

        expect(asFragment()).toMatchSnapshot();
    });
});
