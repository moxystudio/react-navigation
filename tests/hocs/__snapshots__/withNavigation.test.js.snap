// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withNavigation HOC should provide navigation 1`] = `
<DocumentFragment>
  {"drawer":{"isOpen":"false","toggle":"function","open":"function","close":"function"}}
</DocumentFragment>
`;
